### PR TITLE
Add limit and offset Table props for use with pagination

### PR DESCRIPTION
**Pull request by Croug**
## Description

I've added a limit and offset property to the Table component so it can be paginated easier via the Pagination component or otherwise. This also means that when sorting the table, all data will be sorted, not just visible data. Currently only implemented in Svelte.

Implements #161 

## Checklist:

- [x] Is this a new feature
- [x] Have you submitted new feature or bugfix issue? Unless it's a small bugfix, we generally prefer to track PRs against an issue which starts a dialogue to give us context.
- [x] Tests passing?
- [ ] Have you updated the docs? These live in [site/docs](https://github.com/AgnosticUI/agnosticui/tree/master/site/docs)

### CODE REVIEW

**Comment by Croug** [comment]
Also may be useful to have an on:sort hook now, as some users may want to go back to the first page on sort

File: agnostic-svelte/src/lib/components/Table/Table.stories.js [hunk 1]
```diff
@@ -138,3 +138,9 @@ CaptionBottom.args = { ...defaultArgs, captionPosition: "bottom" };
 
 export const CaptionEnd = Template.bind({});
 CaptionEnd.args = { ...defaultArgs, captionPosition: "end" };
+
+export const LimitFive = Template.bind({});
+LimitFive.args = { ...defaultArgs, limit: 5 };
+
+export const OffsetFive = Template.bind({});
```
Comment (roblevintennis) [inline agnostic-svelte/src/lib/components/Table/Table.stories.js:145]: I think there's 8 rows in the fixture right? So maybe we should have a test with both an offset of 2 and a limit of 3 or something just to test using both props with the size of dataset?

**Review by roblevintennis — CHANGES_REQUESTED** [review]
I like the simplicity and of course now that I'm seeing the code I better understand what you shooting for 👍🏽

Mind just adding the one more spec with both props 🙏🏽 thanks!

**Comment by roblevintennis** [comment]
I'm assuming that you're planning to get this working in your app and tie the pagination + table. If you make a dead simple codesanbox I can eventually add that as a playground linked from the docs (see the /playground directory). Totally optional but would be super helpful if you feel compelled 😉

**Comment by Croug** [comment]
Sure thing I'll get on those changes either tonight or tomorrow, and I'll put something together for the playground.

File: agnostic-svelte/src/lib/components/Table/Table.stories.js (hunk 1 above)
Comment (Croug) [reply agnostic-svelte/src/lib/components/Table/Table.stories.js:145]: Alright added an Offset2Limit3 spec

**Comment by Croug** [comment]
Alright created the sandbox, I couldn't test it with the table itself as the playground uses the official agnosticui-svelte package, but I was able to test everything else in there, so we should be good, I'll work on the on:sort hook tomorrow

**Comment by roblevintennis** [comment]
This appears to do as intended:

image

Thanks @Croug this is hugely helpful! Looking forward to the event dispatch and seeing the playground evolve!

___

Not sure if you might have done something similar locally, but here's a little trick I use to test these sorts of things locally before published…

```shell
cd agnostic-svelte
npm run package
cd package && npm pack && mv agnostic-svelte-1.1.18.tgz ..
```

And then from the playground

```shell
npm i ~/workspace/opensource/agnosticui/agnostic-svelte/agnostic-svelte-1.1.18.tgz
```

Running `npm pack` generates the tarball as I understand the exact same as if you'd done `npm publish`. And then you can test it before publishing locally. I've found it's much more reliable then `npm link`.

And the wonderful result given your contribution!

image

That's just a single page but paginating around after clicking the various column sorts appears to be working 💪 🙌  It's quite cool to see this wired up @Croug 🙏 I'm very excited to also get the `on:sort` event dispatcher too!

I think it probably makes sense for me to wait until `on:sort` is in before doing another patch bump and publish (but I can't wait! 🕺 🔥 )

**Pull request merged**